### PR TITLE
fix(npd): point node-problem-detector at the real kubelet kubeconfig

### PR DESCRIPTION
**Pull request by dhia-gharsallaoui**
## What

`node-problem-detector` (NPD) was configured with a doubled path segment for the kubelet kubeconfig — `/var/lib/kubelet/kubelet/kubeconfig` instead of `/var/lib/kubelet/kubeconfig`. As a result NPD's in-cluster client panics on startup on **every** flex node, and systemd restarts it indefinitely.

Observed on a node up ~6 days (restart counter 46,793 and climbing):

```
panic: stat /var/lib/kubelet/kubelet/kubeconfig: no such file or directory
  problemclient.NewClientOrDie(...) /src/pkg/exporters/k8sexporter/problemclient/problem_client.go:72
node-problem-detector.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
node-problem-detector.service: Scheduled restart job, restart counter is at 46793.
```

## Root cause

`pkg/npd/start.go` declared its own constant:

```go
const (
    KubeletKubeconfigPath = "/var/lib/kubelet/kubelet/kubeconfig"  // doubled "kubelet"
    ...
)
```

The kubelet actually writes its kubeconfig to /var/lib/kubelet/kubeconfig. The imported agent library already defines this correctly and wires the kubelet to it:

- github.com/Azure/unbounded/pkg/agent/goalstates → KubeletKubeconfigPath = "/var/lib/kubelet/kubeconfig"
- used by the kubelet phase at pkg/agent/phases/nodestart/kubelet.go

So the npd package's separate copy of the constant was both redundant and wrong.

## Change

- Remove the local KubeletKubeconfigPath constant and use the canonical goalstates.KubeletKubeconfigPath (already a transitive dependency via pkg/agent/phases). This fixes the path and removes the duplicate source of truth so it can't drift again.
- Add regression tests for the pkg/npd package (previously had none):
  - Start() wires the canonical, non-doubled path.
  - The rendered NPD systemd unit's ExecStart auth= parameter resolves to /var/lib/kubelet/kubeconfig.

## Testing

- go build ./..., go vet ./pkg/npd/..., gofmt — all clean.
- New tests pass; verified non-vacuous (they fail against the pre-fix code with kubeconfigPath = "/var/lib/kubelet/kubelet/kubeconfig", want "/var/lib/kubelet/kubeconfig").

## Impact

Fixes NPD for all future flex-node bootstraps. Already-running nodes need a redeploy of the agent (or an in-place fix of the unit inside the kube1 machine) to stop the existing crash-loop.

### CODE REVIEW

**Comment by dhia-gharsallaoui** [comment]
@microsoft-github-policy-service agree

File: pkg/npd/start.go [hunk 1]
```diff
@@ -14,15 +14,15 @@ import (
...
 	"github.com/Azure/unbounded/pkg/agent/phases"
 )
 
 //go:embed assets/node-problem-detector.service
 var serviceTemplate string
 
 const (
-	KubeletKubeconfigPath = "/var/lib/kubelet/kubelet/kubeconfig"
-	systemdUnitNPD        = "node-problem-detector.service"
+	systemdUnitNPD = "node-problem-detector.service"
```
[the remark this answers is not in the document]
Comment (bcho) [reply pkg/npd/start.go:25]: Removing from current settings is accpetable

**Comment by bcho** [comment]
Thank you for catching the issue, it was introduced by previous refactor. I have included your fix and test validation in #198
